File: packages/native-modules/android/src/main/java/expo/modules/nativeuiorgnativemodules/contextmenu/NativeUiOrgContextMenuModule.kt
package expo.modules.nativeuiorgnativemodules

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class NativeUiOrgContextMenuModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("NativeUiOrgContextMenu")

    // Define the native view component (used for iOS)
    View(NativeUiOrgContextMenuView::class) {
      // Props
      Prop("menuItems") { view: NativeUiOrgContextMenuView, items: List<Map<String, Any>> ->
        view.setMenuItems(items)
      }
      
      // Events
      Events("onMenuOpen", "onMenuClose", "onItemPress")
    }
  }
}

File: packages/native-modules/ios/context-menu/NativeUiOrgContextMenuModule.swift
import ExpoModulesCore

public class NativeUiOrgContextMenuModule: Module {
  public func definition() -> ModuleDefinition {
    Name("NativeUiOrgContextMenu")

    // Define the native view component
    View(NativeUiOrgContextMenuView.self) {
      // Props
      Prop("menuItems") { (view: NativeUiOrgContextMenuView, items: [[String: Any]]) in
        view.setMenuItems(items)
      }
      
      // Events
      Events("onMenuOpen", "onMenuClose", "onItemPress")
    }
  }
}

